### ISSUE TITLE
[no-unused-vars] Reports errors on TypeScript modules and namespaces

### ISSUE THREAD

**Issue by boyanio** [typescript-eslint/typescript-eslint#2346]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-vars": "error"
  }
}
```

```TS
module people {
  export interface Person {
    name: string;
  }
}
```

**Expected Result**

No errors

**Actual Result**

I get an error that `people` is defined, but not used. This is true for this file, but it is used in other files.

**Additional Info**

No errors in eslint. I read issue #1856 but I am not sure it is related to it

**Versions**

| package                                   | version |
| ----------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.7.1` |
| `@typescript-eslint/parser`               | `3.7.1` |
| `TypeScript`                              | `3.9.2` |
| `ESLint`                                  | `7.5.0` |
| `node`                                    | `12.16.3` |
| `npm`                                     | `6.14.4` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/1856

Is about _all_ issues to do with rules that use scope analysis. Which is why it's a meta issue and so many issues are rolled up into it.

The v4 rc is out on NPM with a new scope analyser. Give it a go.

The npm tag is `rc-v4`
